build_mt.py

```diff
@@ -1,599 +1,827 @@
-    mask_lc_this_info = (((com_dist_this_info > chi_this) & (com_dist_this_info <= chi_prev))) & (eligibility_this_info) #| \
-    #((com_dist_prev_main_this_info > chi_this) & (com_dist_prev_main_this_info <= chi_prev))) & (eligibility_this_info)
-    
-    mask_lc_this_noinfo = ((com_dist_this_noinfo >= chi_this - delta_chi_old/2.) & \
-                           (com_dist_this_noinfo < chi_this + delta_chi/2.)) & (eligibility_this_noinfo)
-    # why not just remove those at the beginning
-    
-    # percentage of objects that are part of this or previous snapshot
-    print("percentage of halos in light cone with and without progenitor info = ", \
-          np.sum(mask_lc_this_info)/len(mask_lc_this_info)*100., np.sum(mask_lc_this_noinfo)/len(mask_lc_this_noinfo)*100.)
-
-    # select halos with mt info that have had a light cone crossing
-    pos_this_info_lc = pos_this_info[mask_lc_this_info]
-    com_dist_this_info_lc = com_dist_this_info[mask_lc_this_info]
-    pos_prev_main_this_info_lc = pos_prev_main_this_info[mask_lc_this_info]
-    com_dist_prev_main_this_info_lc = com_dist_prev_main_this_info[mask_lc_this_info]
-    halo_ind_prev_main_this_info_lc = halo_ind_prev_main_this_info[mask_lc_this_info]
-    halo_ind_this_info_lc = halo_ind_this_info[mask_lc_this_info]
-    eligibility_this_info_lc = eligibility_this_info[mask_lc_this_info]
-    if 'Progenitors' in fields_mt:
-        start_progs_this_info_lc = start_progs_this_info[mask_lc_this_info]
-        num_progs_this_info_lc = num_progs_this_info[mask_lc_this_info]
+    # assert np.sum((chi_star > chi1) | (chi_star < chi2)) == 0, "Solution is out of bounds"
 
+    return chi_star, pos_star, vel_star, bool_star
 
-    
-    # START TESTING
-    '''
-    x_min = 0.
-    x_max = x_min+10.
-    
-    x = pos_this_info_lc[:,0]
-    choice = (x > x_min) & (x < x_max)
-    
-    y = pos_this_info_lc[choice,1]
-    z = pos_this_info_lc[choice,2]
 
-    plt.figure(1)
-    plt.scatter(y,z,color='dodgerblue',s=0.1,label='current objects')
+def get_one_header(merger_dir):
+    '''Get an example header by looking at one association file in a merger directory'''
+    fn = list(merger_dir.glob('associations*.asdf'))[0]
+    with asdf.open(fn) as af:
+        header = af['header']
+    return header
+
+
+def main(sim_name, z_start, z_stop, resume=False, plot=False):
+    # speed of light
+    c = 299792.458  # km/s
+    
+    merger_parent = Path("/mnt/gosling2/bigsims/merger")
+    merger_dir = merger_parent / sim_name
+    header = get_one_header(merger_dir)
+    
+    # simulation parameters
+    Lbox = header['BoxSize']
+    # location of the LC origins in Mpc/h
+    origins = np.array(header['LightConeOrigins']).reshape(-1,3)
+    origin = origins[0]
+
+    # directory where we save the final outputs
+    cat_lc_dir = Path("/mnt/gosling1/boryanah/light_cone_catalog/") / sim_name / "halos_light_cones/"
+    os.makedirs(cat_lc_dir, exist_ok=True)
```

Comment:
The `exist_ok` arg is great; lets you avoid checking if the dir exists.

Can also write: `cat_lc_dir.mkdir(parent=True, exist_ok=True)`, but that's actually more verbose.